test/unit/ButtonTokenWethRouter.ts

```diff
@@ -0,0 +1,148 @@
+import { ethers, waffle } from 'hardhat'
+import { Contract, Signer, BigNumber } from 'ethers'
+import { expect } from 'chai'
+const { loadFixture } = waffle
+
+const PRICE_DECIMALS = 8
+const NAME = 'Button Bitcoin'
+const SYMBOL = 'BTN-BTC'
+
+interface TestContext {
+  accounts: Signer[]
+  buttonToken: Contract
+  mockOracle: Contract
+  weth: Contract
+  router: Contract
+}
+
+const toOracleValue = (v: string): BigNumber =>
+  ethers.utils.parseUnits(v, PRICE_DECIMALS)
+
+async function fixture(): Promise<TestContext> {
+  const accounts = await ethers.getSigners()
+
+  const wethFactory = await ethers.getContractFactory('WETH9')
+  const weth = await wethFactory.deploy()
+
+  const oracleFactory = await ethers.getContractFactory('MockOracle')
+  const mockOracle = await oracleFactory.deploy()
+  await mockOracle.setData(toOracleValue('10000'), true)
+
+  const buttonTokenFactory = await ethers.getContractFactory('ButtonToken')
+  const buttonToken = await buttonTokenFactory.deploy()
+
+  buttonToken.initialize(weth.address, NAME, SYMBOL, mockOracle.address)
+
+  const routerFactory = await ethers.getContractFactory('ButtonTokenWethRouter')
+  const router = await routerFactory.deploy(weth.address)
+
+  return {
+    accounts,
+    buttonToken,
+    mockOracle,
+    weth,
+    router,
+  }
+}
+
+describe('ButtonTokenWethRouter', () => {
+  it('should instantiate router', async function () {
+    const { router } = await loadFixture(fixture)
+
+    expect(router.address).to.exist
+  })
+
+  it('should deposit ETH', async function () {
+    const { router, weth, accounts, buttonToken } = await loadFixture(fixture)
+    const [user] = accounts
+
+    const depositAmount = ethers.utils.parseEther('5')
+    await expect(router.deposit(buttonToken.address, { value: depositAmount }))
+      .to.emit(weth, 'Deposit')
+      .withArgs(router.address, depositAmount)
+      .to.emit(buttonToken, 'Transfer')
+      .withArgs(
+        ethers.constants.AddressZero,
+        await user.getAddress(),
+        depositAmount.mul('10000'),
+      )
+
+    expect(await buttonToken.balanceOf(await user.getAddress())).to.equal(
+      depositAmount.mul('10000'),
+    )
+  })
+
+  it('should burn button tokens', async function () {
+    const { router, accounts, buttonToken } = await loadFixture(fixture)
+    const [user] = accounts
+
+    const depositAmount = ethers.utils.parseEther('5')
+    await router.deposit(buttonToken.address, { value: depositAmount })
+
+    const userStartingBalance = await user.getBalance()
+    await buttonToken.approve(router.address, ethers.utils.parseEther('50000'))
```

Comment:
readability: shouldn't this be 2500? because down in line 86 we only burn 2500

Reply:
sure yup